prometheus_client/metrics.py

```diff
@@ -36,18 +37,32 @@ def _build_full_name(metric_type, name, namespace, subsystem, unit):
     return full_name
 
 
+def _validate_labelname(l):
+    if not METRIC_LABEL_NAME_RE.match(l):
+        raise ValueError('Invalid label metric name: ' + l)
+    if RESERVED_METRIC_LABEL_NAME_RE.match(l):
+        raise ValueError('Reserved label metric name: ' + l)
+
+
 def _validate_labelnames(cls, labelnames):
     labelnames = tuple(labelnames)
     for l in labelnames:
-        if not METRIC_LABEL_NAME_RE.match(l):
-            raise ValueError('Invalid label metric name: ' + l)
-        if RESERVED_METRIC_LABEL_NAME_RE.match(l):
-            raise ValueError('Reserved label metric name: ' + l)
+        _validate_labelname(l)
         if l in cls._reserved_labelnames:
             raise ValueError('Reserved label metric name: ' + l)
     return labelnames
 
 
+def _validate_exemplar(exemplar):
+    runes = 0
+    for k, v in exemplar.items():
+        _validate_labelname(k)
+        runes += len(k.encode('utf-8'))
```

Comment:
This is measuring bytes, not UTF-8 characters. `len(k)` is what you want.

Reply:
:+1: thanks, I also added a test with a Japanese character.